integration/mediation-tests/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/servers/GreenMailServer.java

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.esb.integration.common.utils.servers;
+
+import com.icegreen.greenmail.user.GreenMailUser;
+import com.icegreen.greenmail.util.GreenMail;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.util.Properties;
+import javax.mail.Flags;
+import javax.mail.Folder;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.Session;
+import javax.mail.Store;
+import javax.mail.URLName;
+import javax.mail.search.AndTerm;
+import javax.mail.search.BodyTerm;
+import javax.mail.search.SearchTerm;
+import javax.mail.search.SubjectTerm;
+
+/**
+ * Util class for GreenMail Server.
+ */
+public class GreenMailServer {
+    protected static Log log = LogFactory.getLog(GreenMailServer.class);
+    private static GreenMail greenMail;
+    private static GreenMailUser primaryUser;
+    private static final String EMAIL_INBOX = "INBOX";
+    private static final String EMAIL_TRASH = "Trash";
+    private static final int WAIT_TIME_MS = 180 * 1000;
+    private static final String USER_EMAIL = "wso2@localhost";
+    private static final String USER_LOGIN = "wso2";
+    private static final String USER_PW = "wso2";
+    private static final String PROTOCOL_IMAP = "imap";
+    private static final String PROTOCOL_POP3 = "pop3";
+    private static final String BIND_ADDRESS = "127.0.0.1";
+
+    /**
+     * Start the server and add the user.
+     */
+    public void startServer() {
+        greenMail = new GreenMail();
+        greenMail.start();
+        primaryUser = greenMail.setUser(USER_EMAIL, USER_LOGIN, USER_PW);
+        log.info("GreenMail Server started and user added!");
+    }
+
+    /**
+     * Stop server.
+     */
+    public void stopServer() {
+        greenMail.stop();
```

Comment:
It seems `stopServer() `should be static.

Reply:
Fixed with https://github.com/wso2/product-ei/pull/2148